### ISSUE TITLE
Cinnamenu: Feature Request - Action in context menu to edit desktop file

### ISSUE THREAD

**Issue by jonath92** [linuxmint/cinnamon-spices-applets#4631]
@fredcw could you image to add an option "edit desktop file" to the context menu?

I know this is more an advanced feature but it would definitely be useful for me and maybe also for others. Sometimes I want to add arguments to the "Exec" in the desktop file, change the icon or just want to know the exec. Currently it is a little bit cumbersome to figure out the respective desktop file.

**Comment by rcalixte** [comment]
@jonath92 I had the same issue recently and I found that the entries in the Menu Editor (cinnamon-menu-editor) were great. It might be nice to search in this app rather than guess the right category similar to the recent changes made in the Keyboard Shortcuts interface. (NB, this is just an idea I'm throwing into the ether for now. 😅) Has the Menu Editor not met your needs?

**Comment by jonath92** [comment]
Hi @rcalixte I am not a big Fan of the menu Editor because of the missing search function. With my suggestion above I could find desktop.files significantly faster.

**Comment by fredcw** [comment]
@jonath92 Someone else asked for this as well. While I think some people would find it useful and I'd probably find it useful myself occasionally, I don't think it works with my design philosophy for the applet and such a technical option would look out of place. I don't think that an average computer user should be expected to use the command line or even know what a .desktop file is, everything should "just work" as it were while allowing for "under the hood" stuff in linux generally for those who want it. In other words, I want all parts of Cinnamenu's UI to be understandable to the average non-technical user. I guess there's a place for an applet for more advanced users but that not really what I have in mind for cinnamenu. I could add this as a hidden option but to be honest, that's more work and more complexity for an applet that's already quite complex :p

**Comment by rcalixte** [comment]
It sounds like adding a search option to the Menu Editor is the compromise here? I'll volunteer to give it a shot with this open Issue as the base if so.

**Comment by fredcw** [comment]
Ah, I have an idea. I could implement it as key combination - highlight the app and press a key combo. That should be fairly easy to do.

**Comment by Gr3q** [comment]
I don't think that's good UX, just having hidden functionality with no UI to link to/note the keybinding

**Comment by fredcw** [comment]
@rcalixte If you want to get into the code of the menu editor, there are some long standing bugs that could be fixed https://github.com/linuxmint/cinnamon/issues?q=is%3Aissue+is%3Aopen+menu+editor+in%3Atitle

I suspect the bugs may be more complicated to fix than may initially appear though.

**Comment by fredcw** [comment]
@Gr3q I agree, I could document the keybindings on the applet page. My main concern here is to not make the UI too complex for normal users.

**Comment by rcalixte** [comment]
> I suspect the bugs may be more complicated to fix than may initially appear though.

Is that ever not true? 😅

**Comment by rcalixte** [comment]
> @Gr3q I agree, I could document the keybindings on the applet page. My main concern here is to not make the UI too complex for normal users.

Why not have a setting for adjusting the key binding? Same as other applets.

**Comment by fredcw** [comment]
@jonath92 Would opening cinnamon's launcher properties dialogue (same as clicking "properties" in the menu editor) suffice or do you think there should be an option to open the .desktop file in an editor as well?

**Comment by Drugwash2** [comment]
> do you think there should be an option to open the .desktop file in an editor

I believe that was the request in the very first post of this issue. All the talk in-between is just nonsense.
There are applications that set a great deal of info in their desktop files, and some of that info might need to be fixed/tweaked according to user's needs. I myself found the need to edit desktop files and found it extremely hard to do when one doesn't even know what those files are and/or where they are located.

So yeah, an option to directly edit `.desktop` files would be a very welcome addition, but not (only) to some applet - as is the case here - but to the OS itself, for every desktop environment. Well, at least that's my personal opinion.

**Comment by fredcw** [comment]
I'll add the following keybindings unless anyone has better suggestions. I used Alt-Enter because that's also used in Nemo and windows for file properties.

* `Shift-Enter` Run currently highlighted application as root user.
 * `Alt-Enter` - Open launcher properties dialogue of currently highlighted application.
 * `Ctrl-d` - Open .desktop file of currently highlighted app in default text editor (opens file in user mode.)

**Comment by Drugwash2** [comment]
> unless anyone has better suggestions.

Anything for mouse-centric people...? Such as a visible option that people could actually acknowledge and click if/when needed?
With a bad keyboard and/or cats in the house that could trigger any kind of unwanted events by walking on the keyboard I'd say a context-menu item might be a valid alternative approach.

**Comment by jonath92** [comment]
@fredcw thank you very much :-).  That's a good solution.

**Comment by stephan-dev** [comment]
TL;DR :  I think this very cool feature could deserve more that the `Ctrl-d` shortcut, so that even non-technical users can find it !

--

I just discovered that feature, which I've wanted on linux since 2016. It's really cool and very important !
Sorry I have to butt in : I'm not asking for any change, I'll just debate the idea that a normal user shouldn't have to know about `.desktop` files.
I started trying Ubuntu in 2016, hoping to find better than Windows : with Windows, over the years, the more you level up, the more the updates dumb everything down, adding what I call the "dumb it down layer of abstraction".
Of course for lots of things, linux was better. But for program launchers, I realized that Ubuntu (Unity) was not only not better than Windows, but *worse*.
Not everyone is a new user of computers. Computer illiteracy diminishes constantly.
Thing is, on Windows, you can right-click a launcher aka "shortcut properties", and then you find what program you're really launching. I suppose Windows did this because the list of scenarios where a non-technical user needs this is quite big. You need that quite often. (on top of my head (linux example) : oops, I installed 2 versions of the same program, normal and flatpak, but now I don't know which is which!)

When I discovered Ubuntu with Unity DE, I found it even more insulting than Windows, that ALL the real names of the program had to be changed to a "friendly" name, and much worse, there was no GUI way (like on Windows) to find out what one was _really_ launching.

I'm ranting, but it's even worse when you realize that even once you found the `/usr/share/applications` folder, the Nautilus file manager would **still** hide (in many cases) the real .desktop filenames! - AFAIK, it's still the case with Nemo in 2023.

So, I think this very cool feature could deserve more that the `Ctrl-d` shortcut, so that even non-technical users can find it !

**Comment by Drugwash2** [comment]
> I'll just debate the idea that a normal user shouldn't have to know about `.desktop` files.

There is compromise in each and every aspect of our lives, and most of the times it is not done by our own free will but imposed on us by others who think they know better. You are absolutely right about the dumbing down but not only in regard to Windows, or computing generally.

Fortunately there (still) are alternatives for some of the things that by default are being dumbed down. I found Windows Commander (later on renamed as Total Commander) just a couple years after starting using Windows in 1997. I never ever used Windows Explorer since. In Linux (Mint Cinnamon) I had desperately tried to get Krusader work until I found Double Commander. Never ever used Nemo since.

People should just not give up, and keep searching for alternatives when the defaults are unacceptable.

![Screenshot from 2023-06-13 04-25-41](https://github.com/linuxmint/cinnamon-spices-applets/assets/29117024/59a83b07-3cfe-4fb4-9c24-c49e59dcdfad)

**Comment by fredcw** [comment]
@stephan-dev I agree with your observations. I hated it when windows introduced shortcuts, i.e. files that weren't what they appeared to be. I thought it was a completely unnecessary complication that was a substitute for a tiny bit of computer literacy. Even worse was when they started hiding file extensions by default which seemed like nothing but a security risk.

>Nautilus file manager would still hide (in many cases) the real .desktop filenames! - AFAIK, it's still the case with Nemo 

Yes I agree. In nemo, .desktop files should be shown for what they are and only used as links on the desktop. People don't go to Nemo to launch applications from .desktop files. Maybe someone should open an issue for that?

I should add a tip on the tips tab in cinnamenu config about ctrl d & p which I didn't think of before.

**Comment by stephan-dev** [comment]
@Drugwash2 Yes, as for me, I use Midnight Commander, which I like a lot (and Chost Commander on Android) (but I use Nemo too)  

@fredcw   
> Windows hiding file extensions

Yes good example, where will they stop the dumbing down ? Also, huge buttons for everyone because seamless experience with tablets / touchscreens => Here we are, back in kindergarten ! 

> Open an issue about Nemo incorrectly showing .desktop files 

(like Nautilus does...) Good idea.  

> add a tip on the tips tab in cinnamenu config about ctrl d & p which I didn't think of before  

Yes I think that would be very useful ! In theory, only geeks needs this, in practice, the launchers can often be confusing, like when a local program and its flatpak version have exactly the same name.
